### DISCUSSION TITLE
I put in the wrong email, how do I reset it?

### DISCUSSION THREAD

**Discussion by realterragenesis** [community/community#22127] [New to GitHub]
<p>I put in the wrong email and now I can’t push to my repository. How do I reset it?</p>

**Comment by paulgureghian** [comment]
<p>Did you use the ‘git config --global’  syntax ? or just at the prompt when committing / pushing ?</p>

**Comment by realterragenesis** [comment]
<p>At the prompt when commiting/pushing</p>

**Comment by paulgureghian** [comment]
<p>Just try the commit / push again.</p>
<p>Or permanently store your credentials with ‘config --global’  or ‘credential-store’ option.</p>

**Comment by realterragenesis** [comment]
<p>Wait no, I think I also used the --global syntax as well</p>

**Comment by realterragenesis** [comment]
<p>By that I mean when it prompted me it told me to use --global</p>

**Comment by paulgureghian** [comment]
<p>Then just run global again and change the credentials</p>

**Comment by realterragenesis** [comment]
<p>I tried that, it didn’t work (unless I’m doing something wrong)</p>

**Comment by airtower-luna** [comment]
<p>Might any of the mail addresses in your commits be ones that you’ve configured Github to keep private? In general it’ll be easier to help you if you post the exact error message (you can censor any private details).</p>

**Comment by realterragenesis** [comment]
<p>Sure, give me a second</p>

**Comment by realterragenesis** [comment]
<p><a href="mailto:git@github.com">git@github.com</a>: Permission denied (publickey).<br>
fatal: Could not read from remote repository.</p>
<p>Please make sure you have the correct access rights<br>
and the repository exists.</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>That error message has nothing to do with mail addresses, it means Github doesn’t accept your SSH key for login. This can have a bunch of reasons:</p>
<ul>
<li>There’s not SSH key available on your computer.</li>
<li>You don’t have the right key loaded into the SSH agent (if it is password protected, or has a non-default file name).</li>
<li>You don’t have the key for your current computer configured on your Github account.</li>
</ul>
<p>I recommend taking a look at the documentation on <a href="https://help.github.com/en/github/authenticating-to-github/connecting-to-github-with-ssh" rel="nofollow noopener">Connecting to GitHub with SSH</a>, it has pages for each of those possible problems.</p>

**Comment by realterragenesis** [comment]
<p>Thank you, it worked! However it appears I have a new problem. When I try to push it tells me to pull, but when I try to pull it returns this error:</p>
<blockquote>
<p>fatal: refusing to merge unrelated histories</p>
</blockquote>

**Comment by airtower-luna** [comment]
<p>That means your local repository has no commits in common with the remote one. Did you change the first commit by any chance (e.g. using <code>git rebase</code> or <code>git commit --amend</code>)?</p>

**Comment by realterragenesis** [comment]
<p>I think I used git rebase once.</p>

**Comment by airtower-luna** [comment]
<p>If that rebase changed the first (“initial”) commit in the repository that’s probably the reason. In general rebasing history that’s shared will have funny results.</p>
<p>There are a few things you can do to resolve this. But all of them carry a risk of messing up and losing data, so be very careful! :warning:</p>
<ul>
<li>If you are absolutely sure you want to keep the local version and are fine with losing <em>everything</em> on the branch you’re pushing to, you can use <code>git push --force</code>. That will replace the remote branch with the local one.</li>
<li>If you want to keep the remote history and add a few commits from the local branch, you can create a new local branch from the remote branch, and use <code>git cherry-pick</code> to copy the commits you want. Make absolutely sure you got everything you want before deleting the old local branch!</li>
<li>For completeness sake I’ll mention that there is a <code>git merge</code> option to allow merging unrelated histories, but that is almost guaranteed to be the wrong choice for data that is supposed to have a common history.</li>
</ul>

**Comment by realterragenesis** [comment]
<p>Thank you, it worked! Thank you for all the help.</p>
